src/java.base/windows/native/libnet/NetworkInterface_winXP.c

```diff
@@ -274,20 +268,23 @@ int getAllInterfacesAndAddresses (JNIEnv *env, netif **netifPP)
     // Retrieve IPv4 addresses with the IP Helper API
     curr = *netifPP;
     ret = lookupIPAddrTable(env, &tableP);
-    if (ret < 0) {
+    if (ret == -1) {
+      free_netif(*netifPP);
       return -1;
+    } else if (ret == -2) {
+        // Clear the exception and continue.
+        if ((*env)->ExceptionCheck(env)) {
+            (*env)->ExceptionClear(env);
+        }
+        tableP = NULL;
     }
     while (curr != NULL) {
         netaddr *netaddrP;
         ret = enumAddresses_win_ipaddrtable(env, curr, &netaddrP, tableP);
-        if (ret == -1) {
+        if (ret < 0) {
+            free_netif(*netifPP);
```

Comment:
seems like the right thing to do, but *netifPP may not be properly assigned a valid pointer

Reply:
`netifPP` is always assigned by `enumInterfaces` in nonnegative return cases. If `enumInterfaces` returns a negative value, this function (`getAllInterfacesAndAddresses`) aborts much earlier.